next/app/sitemap.ts

```diff
@@ -0,0 +1,70 @@
+import { MetadataRoute } from "next";
+
+import { drupal } from "@/lib/drupal/drupal-client";
+import { GET_SITEMAP_NODES } from "@/lib/graphql/queries";
+import {
+  addSitemapLanguageVersionsOfFrontpage,
+  addSitemapLanguageVersionsOfNode,
+  makePathAbsolute,
+} from "@/lib/utils";
+
+import siteConfig from "@/site.config";
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  // Get all languages from site config:
+  const languages = Object.keys(siteConfig.locales);
+  // Initialize the sitemap:
+  let sitemap = [];
+
+  // For each language, start fetching the nodes for the sitemap:
+  for (const lang of languages) {
+    let page = 0;
+    let totalItems = 0;
+    // Initialize the page size to 0, we will get the actual value from the first request:
+    let pageSize = 0;
+
+    do {
+      const data = await drupal.doGraphQlRequest(
+        GET_SITEMAP_NODES,
+        {
+          page: page,
+          langcode: lang,
+        },
+        false, // We only want published nodes, so we disable authentication.
```

Comment:
We need to disable auth because we only want published nodes in the sitemap

(also, in my local at least, it errored out sometimes, I wonder if something in drupalClient is not compatible with app router). This way it works.